packages/cli/core/src/commands/ui/deploy.spec.ts

```diff
@@ -465,6 +473,69 @@ describe('ui:deploy', () => {
         }
       );
 
+    test
+      .stdout()
+      .stderr()
+      .do(() => {
+        doMockConfirm();
+        mockedConfirm.mockResolvedValueOnce(true);
+        mockHostedPageList.mockResolvedValueOnce({
+          items: [{name: validJsonConfig.name, id: 'thisistheid'}],
+        });
+      })
+      .command(['ui:deploy'])
+      .it(
+        'should call hostedPages.list when no page id argument is passed and a page with the same already exists',
+        () => {
+          expect(mockHostedPageList).toHaveBeenCalledWith(
+            expect.objectContaining({
+              filter: validJsonConfig.name,
+              perPage: 1000,
+            })
+          );
+        }
+      );
+
+    test
+      .stdout()
+      .stderr()
+      .do(() => {
+        doMockConfirm();
+        mockedConfirm.mockResolvedValueOnce(true);
+        mockHostedPageList.mockResolvedValueOnce({
+          items: [{name: validJsonConfig.name, id: 'thisistheid'}],
+        });
+      })
+      .command(['ui:deploy'])
+      .it(
+        'should call hostedPages.update when no page id argument is passed, a page with the same already exists, and the user confirms the overwrite',
+        () => {
+          expect(mockHostedPageUpdate).toHaveBeenCalledWith({
+            ...expectedHostedPage,
+            id: 'thisistheid',
+          });
+        }
+      );
+
+    test
+      .stdout()
+      .stderr()
+      .do(() => {
+        doMockConfirm();
+        mockedConfirm.mockResolvedValueOnce(false);
+        mockHostedPageList.mockResolvedValueOnce({
+          items: [{name: validJsonConfig.name, id: 'thisistheid'}],
+        });
+      })
+      .command(['ui:deploy'])
+      .catch((err) => expect(err).toMatchSnapshot())
+      .it(
+        'should not call hostedPages.update when no page id argument is passed, a page with the same already exists, and the user decline the overwrite',
```

Comment:
"when no page id argument is passed" - Isn't the `id` being passed above? the only diff i can see with the previous tests is the `mockedConfirm` is passed a negative value.

Reply:
"the user decline the overwrite" -> "the user declines the overwrite"

Reply:
> "when no page id argument is passed" - Isn't the `id` being passed above? the only diff i can see with the previous tests is the `mockedConfirm` is passed a negative value.

No actually, to see what the 'user' has input, assuming no interaction, you can check the `.command` function call
This function essentially is 'call the CLI with this'.
https://oclif.io/docs/testing